src/components/containers/TraceAccordion.js

```diff
@@ -23,14 +23,16 @@ class TraceAccordion extends Component {
   setLocals(props, context) {
     // we don't want to include analysis transforms when we're in the create panel
     const base = props.canGroup ? context.fullData : context.data;
+    const traceFilterCondition =
+      this.props.traceFilterCondition || (() => true);
 
     this.filteredTracesFullDataPositions = [];
     this.filteredTraces = base.filter((t, i) => {
-      if (props.excludeFits) {
-        return !(t.transforms && t.transforms.every(tr => tr.type === 'fit'));
+      if (traceFilterCondition(t, context.fullData[i])) {
```

Comment:
context.fullData[i]
...well, base could be data or fullData, given this, do we still want to send context.fullData[i]?
or maybe just base[i] is ok?

Reply:
we ARE sending base[i]. The second argument is a special case for Colorbar stuff. Doesn't get looked at by any other filter condition funcs.